kubernetes/config.go

```diff
@@ -5,13 +5,12 @@ import (
 	"path/filepath"
 
 	"github.com/docker/docker/pkg/homedir"
-	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
 // NewKubernetesConfig resolves the path to the desired Kubernetes configuration file, depending
 // environment variable and command line flag.
-func NewKubernetesConfig(configFlag string) (*restclient.Config, error) {
+func NewKubernetesConfig(configFlag string) clientcmd.ClientConfig {
 	kubeConfig := configFlag
 	if kubeConfig == "" {
```

Comment:
The exact same code is also in `cli.WrapCli()` (and looks to be redundant). I suggest extracting this to a function, e.g.

```go
func ConfigPath(kubeconfigPath string) string {
	if kubeconfigPath != "" {
		return kubeconfigPath
	}
	if config := os.Getenv("KUBECONFIG"); config != "" {
		return config
	}
	return filepath.Join(homedir.Get(), ".kube/config")
}
```

Is a non-existing path reason for an error?

Reply:
Well actually ```WrapCli``` calls ```NewKubernetesConfig``` right after the duplicated code so this looks like the code was moved/factorized and not removed…
Fixed!